sklearn/model_selection/split.py

```diff
@@ -0,0 +1,1106 @@
+
+    Due to the high number of test sets (which is the same as the
+    number of samples) this cross validation method can be very costly.
+    For large datasets one should favor KFold, StratifiedKFold or
+    ShuffleSplit.
+
+    Read more in the :ref:`User Guide <split>`.
+
+    Examples
+    --------
+    >>> from sklearn.model_selection import LeaveOneOut
+    >>> X = np.array([[1, 2], [3, 4]])
+    >>> y = np.array([1, 2])
+    >>> loo = LeaveOneOut()
+    >>> print(loo)
+    sklearn.model_selection.split.LeaveOneOut()
+    >>> for train_index, test_index in loo.split(X):
+    ...    print("TRAIN:", train_index, "TEST:", test_index)
+    ...    X_train, X_test = X[train_index], X[test_index]
+    ...    y_train, y_test = y[train_index], y[test_index]
+    ...    print(X_train, X_test, y_train, y_test)
+    TRAIN: [1] TEST: [0]
+    [[3 4]] [[1 2]] [2] [1]
+    TRAIN: [0] TEST: [1]
+    [[1 2]] [[3 4]] [1] [2]
+
+    See also
+    --------
+    LeaveOneLabelOut for splitting the data according to explicit,
+    domain-specific stratification of the dataset.
+    """
+
+    def _pre_split_check(self, X, y=None, labels=None):
+        return
+
+    def _iter_test_indices(self, X, y=None, labels=None):
+        return range(_num_samples(X))
+
+    def __repr__(self):
+        return '%s.%s()' % (
+            self.__class__.__module__,
+            self.__class__.__name__,
+        )
+
+
+class LeavePOut(_PartitionIterator):
+    """Leave-P-Out cross validation iterator
+
+    Provides train/test indices to split data in train test sets. This results
+    in testing on all distinct samples of size p, while the remaining n - p
+    samples form the training set in each iteration.
+
+    Note: ``LeavePOut(p)`` is NOT equivalent to ``KFold(n_folds=n // p)``
+    (``n`` = number of samples) which creates non-overlapping
+    test sets.
+
+    Due to the high number of iterations which grows combinatorically with the
+    number of samples this cross validation method can be very costly. For
+    large datasets one should favor KFold, StratifiedKFold or ShuffleSplit.
+
+    Read more in the :ref:`User Guide <split>`.
+
+    Parameters
+    ----------
+    p : int
+        Size of the test sets.
+
+    Examples
+    --------
+    >>> from sklearn.model_selection import LeavePOut
+    >>> X = np.array([[1, 2], [3, 4], [5, 6], [7, 8]])
+    >>> y = np.array([1, 2, 3, 4])
+    >>> lpo = LeavePOut(2)
+    >>> print(lpo)
+    sklearn.model_selection.split.LeavePOut(p=2)
+    >>> for train_index, test_index in lpo.split(X):
+    ...    print("TRAIN:", train_index, "TEST:", test_index)
+    ...    X_train, X_test = X[train_index], X[test_index]
+    ...    y_train, y_test = y[train_index], y[test_index]
+    TRAIN: [2 3] TEST: [0 1]
+    TRAIN: [1 3] TEST: [0 2]
+    TRAIN: [1 2] TEST: [0 3]
+    TRAIN: [0 3] TEST: [1 2]
+    TRAIN: [0 2] TEST: [1 3]
+    TRAIN: [0 1] TEST: [2 3]
+    """
+
+    def __init__(self, p):
```

Comment:
This is `LeavePOut`, not `LeavePLabelOut`, so it should be `n_samples` but I think that can be confusing... `p` is documented by the class name, fwiw.

On June 22, 2015 6:27:09 PM EDT, Andreas Mueller <notifications@github.com> wrote:
>> +    >>> lpo = LeavePOut(2)
>> +    >>> print(lpo)
>> +    sklearn.model_selection.split.LeavePOut(p=2)
>> +    >>> for train_index, test_index in lpo.split(X):
>> +    ...    print("TRAIN:", train_index, "TEST:", test_index)
>> +    ...    X_train, X_test = X[train_index], X[test_index]
>> +    ...    y_train, y_test = y[train_index], y[test_index]
>> +    TRAIN: [2 3] TEST: [0 1]
>> +    TRAIN: [1 3] TEST: [0 2]
>> +    TRAIN: [1 2] TEST: [0 3]
>> +    TRAIN: [0 3] TEST: [1 2]
>> +    TRAIN: [0 2] TEST: [1 3]
>> +    TRAIN: [0 1] TEST: [2 3]
>> +    """
>> +
>> +    def __init__(self, p):
>
>should p be renamed ``n_labels``?
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/scikit-learn/scikit-learn/pull/4294/files#r32991323

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.